llms/mlx_lm/server.py

```diff
@@ -433,8 +455,11 @@ def handle_chat_completions(self) -> mx.array:
             hasattr(self.tokenizer, "apply_chat_template")
             and self.tokenizer.chat_template
         ):
+            messages = body["messages"]
+            if "raise_exception('System role not supported')" in self.tokenizer.chat_template:
```

Comment:
Could you explain what's happening here?

Reply:
The chat template for Gemma raises an exception if a system role is used. Here, I hard-code searching for the raised exception, and fold in the system message into the first user message. It's very hacky, and is actually not needed for running DSPy. I can remove all the logic related to this hack? Then the PR is just about (float, int) :) Thanks!